### ISSUE TITLE
Link to files on s3

### ISSUE THREAD

**Issue by mpacer** [lheagy/nblibrarian#1]
It would be useful to have a way to get files from an s3 bucket (or other cloud storage service).

I’m having difficulty thinking of how this could be addressed without requiring modifying the format of .jupyter_include to handle general uris (including things like s3://{endpoint}/{bucket}/{key}) but that I’m pretty sure would break a standard .gitignore parsed file.

How strong is the commitment to that abstraction?

**Comment by lheagy** [comment]
Thanks @mpacer, agreed - it would be very nice to branch out to cloud storage rather than just assuming github. I don't have much experience with s3, but here is a rough suggestion

In the `library-config.yml` something like
```yaml
source:
    s3:
        endpoint: {endpoint}
        bucket: {bucket}
```
and then the `.jupyter-include` would be the keys. 

For downloading content, we could then generalize the `content_url` to construct the appropriate uris

https://github.com/lheagy/nblibrarian/blob/16ade94f447f27721711e72943301726487a59ab/nblibrarian/librarian.py#L178-L185 

Does a parsing like this make sense? or do you want to be able to combine multiple endpoints / buckets? 

Basically, if it fits, my idea was to use the `library-config` to point to the location that hosts the notebooks, and then if it works, use the `.jupyter-include` to reference local stuff. 

I am totally open to suggestions 😄
